init.ts

```diff
@@ -0,0 +1,55 @@
+import fs from "node:fs";
+import path from "node:path";
+
+console.log("process.argv:", JSON.stringify(process.argv));
+const [_nodePath, _scriptPath, target, inputsRaw = "{}"] = process.argv;
+
+const inputs = JSON.parse(inputsRaw);
+const config = {
+  $schema: "https://docs.renovatebot.com/renovate-schema.json",
+  extends: [
+    `github>bettermarks/renovate-config${inputs.javascript ? ":javascript" : ""}`,
+    inputs.python ? "github>bettermarks/renovate-config:python" : "",
+    inputs.automerge,
+  ].filter(Boolean),
+};
+console.log(config);
+if (!target || !fs.existsSync(target)) {
+  throw (
+    "first argument target has to be a path to a directory but does not exist, was " +
+    target
+  );
+}
+let configFile = path.join(target, "renovate.json");
+if (!fs.existsSync(configFile) || inputs.force) {
+  fs.writeFileSync(configFile, JSON.stringify(config, null, 2), "utf-8");
+  console.error(`written to ${configFile}`);
+} else {
+  console.error(`NOT modifying existing ${configFile}!`);
+}
+
+let relativeWorkflowFile = ".github/workflows/renovate-config-validator.yml";
+let workflowFile = path.join(target, relativeWorkflowFile);
+if (!fs.existsSync(workflowFile) || inputs.force) {
+  fs.cpSync(relativeWorkflowFile, workflowFile);
+  console.error(`created ${workflowFile}!`);
+} else {
+  console.error(`NOT modifying existing ${workflowFile}!`);
+}
+
+if (inputs.javascript) {
+  const npmrcFile = path.join(target, ".npmrc");
+  const saveExact = `save-exact=true`;
+  const existingContent =
+    fs.existsSync(npmrcFile) && fs.readFileSync(npmrcFile, "utf8");
```

Comment:
Will have a closer look tomorrow - just one thing I saw right now: `save-exact` _can_ be an `.npmrc` option but it could also be configured in `pnpm-workspace.yaml`